### PR TITLE
test_build: Increase coverage for build errors patterns

### PR DESCRIPTION
**Pull request by AthenasJimenez**
Signed-off-by: Athenas Jimenez <athenas.jimenez.gonzalez@intel.com>

### CODE REVIEW

File: tests/test_build.py [hunk 1]
```diff
@@ -367,28 +367,24 @@ def mock_util_call(cmd):
...
+            builderrors = f.readlines()
+            for error in builderrors:
+                if not error.startswith('#'):
+                    input, output = error.strip('\n').split('|')
+                    build.buildreq.buildreqs = set()
+                    build.util.open_auto = mock_open(read_data=input)
+                    build.parse_build_results('testname', 0, fm)
+
+                    self.assertIn(output, build.buildreq.buildreqs)
+                    self.assertGreater(build.must_restart, 0)
```
Comment (bryteise) [inline tests/test_build.py]: I am a bit worried that build.must_restart is 0 since a buildreq was added. Am I missing something?

**Comment by phmccarty** [comment]
There are several new unit test failures. Can you take a look at those?

File: tests/test_build.py (hunk 1 above)
Comment (phmccarty) [reply tests/test_build.py]: It's an `assertGreater`, so it shouldn't be 0 at this point, right?

**Comment by phmccarty** [comment]
@AthenasJimenez I suspect the other test failures are raised because `build.util.call` needs to be restored at the end of the test modified in this PR.

**Comment by AthenasJimenez** [comment]
first try didn't work, so I'll keep looking :/

**Comment by AthenasJimenez** [comment]
Fixed :)

**Comment by bryteise** [comment]
Looks good to me.

File: tests/test_build.py (hunk 1 above)
Comment (bryteise) [reply tests/test_build.py]: Indeed!

**Pull request merged**